include/embedding.hpp

```diff
@@ -113,7 +117,9 @@ class embedding {
     inline int weight(int q) const { return qub_weight[q]; }
 
     //! Get the maximum of all qubit weights
-    inline int max_weight() const { return *std::max_element(std::begin(qub_weight), std::begin(qub_weight) + num_qubits); }
+    inline int max_weight() const {
+        return *std::max_element(std::begin(qub_weight), std::begin(qub_weight) + num_qubits);
```

Comment:
why not `std::end`?

Reply:
because it's got length num_qubits+num_reserved

Reply:
:+1: